src/transactionsFactories/smartContractTransactionsFactory.ts

```diff
@@ -80,25 +84,53 @@ export class SmartContractTransactionsFactory {
 
     createTransactionForExecute(options: {
         sender: IAddress,
-        contractAddress: IAddress,
+        contract: IAddress,
         functionName: string,
         gasLimit: BigNumber.Value,
-        args?: any[]
-    }
-    ): DraftTransaction {
+        args?: any[],
+        nativeTransferAmount?: BigNumber.Value,
+        tokenTransfers?: TokenTransfer[]
+    }): DraftTransaction {
         const args = options.args || [];
-        let parts: string[] = [options.functionName];
+        const tokenTransfer = options.tokenTransfers || [];
+        const nativeTransferAmount = options.nativeTransferAmount ?? 0;
+        const numberOfTokens = tokenTransfer.length;
 
-        const preparedArgs = this.argsToDataParts(args, this.abiRegistry?.getEndpoint(options.functionName));
-        parts = parts.concat(preparedArgs);
+        if (nativeTransferAmount && numberOfTokens) {
+            throw new ErrBadUsage("Can't send both native token and ESDT/NFT tokens");
+        }
+
+        let receiver = options.contract;
+        const tokenComputer = new TokenComputer();
+        let transferArgs: string[] = [];
+
+        if (numberOfTokens === 1) {
+            const transfer = tokenTransfer[0];
+
+            if (tokenComputer.isFungible(transfer.token)) {
+                transferArgs = this.dataArgsBuilder.buildArgsForESDTTransfer(transfer);
+            }
+            else {
+                transferArgs = this.dataArgsBuilder.buildArgsForSingleESDTNFTTransfer(transfer, receiver);
+                receiver = options.sender;
+            }
+        }
+        else if (numberOfTokens > 1) {
```

Comment:
Else, without if (also for py). Also, formatting / (braces & newlines).

Reply:
As it is now, it doesn't properly work without `else if`. The code checks if someone provides a native token amount and custom token transfers but does not check if neither is provided. And it shouldn't check because you can simply call a contract function without doing any transfers. We check if the `numberOfTokens == 1`, but then if we use a simple `else`, and `numberOfTokens = 0`, the else branch get's executed and we don't want that since we don't have any transfers.